stdlib/auth/src/main/ballerina/src/auth/inbound_basic_auth_provider.bal

```diff
@@ -104,24 +119,24 @@ function extractHash(string configValue) returns string {
     return configValue.substring((<int> configValue.indexOf("{")) + 1, stringutils:lastIndexOf(configValue, "}"));
 }
 
-# Reads the password hash for a user.
+# Reads the password hash of a user.
 #
-# + username - Username
-# + return - Password hash read from userstore, or nil if not found
+# + username - Username of the user
+# + return - Password hash read from userstore or else `()` if not found
 function readPassword(string username) returns string {
-    // first read the user id from user->id mapping
-    // read the hashed password from the user-store file, using the user id
+    // First, reads the user ID from user->id mapping.
+    // Then, reads the hashed password from the user-store file using the user ID.
     return getConfigAuthValue(CONFIG_USER_SECTION + "." + username, "password");
 }
 
 function getConfigAuthValue(string instanceId, string property) returns string {
     return config:getAsString(instanceId + "." + property, "");
 }
 
-# Construct an array of groups from the comma separed group string passed.
+# Constructs an array of groups from the given comma-separed string of groups.
 #
-# + groupString - Comma separated string of groups
-# + return - Array of groups, nil if the groups string is empty/nil
+# + groupString - Comma-separated string of groups
+# + return - An array of groups or else () if the groups string is empty/`()`
```

Comment:
```suggestion
# + return - An array of groups or else () if the groups string is empty
```

Reply:
This should reflect string becoming empty and also string becoming `nil`.